### PR TITLE
spoofdpi: init at 0.10.12

### PR DESCRIPTION
**Pull request by s0me1newithhand7s**
## Description of changes
spoofdpi: init 
newer than https://github.com/NixOS/nixpkgs/pull/331972

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by s0me1newithhand7s** [comment]
this commit messages does not look good. i may clear 'em in, but not today..

**Review by ViZiD — CHANGES_REQUESTED** [review]

**Comment by s0me1newithhand7s** [comment]
i'll force-push as soon as i get home.

**Review by ViZiD — APPROVED** [review]

**Review by ViZiD — COMMENTED** [review]
Squash commits

**Comment by s0me1newithhand7s** [comment]
> Squash commits

https://github.com/NixOS/nixpkgs/pull/337795#issuecomment-2314481996
and thx!

**Review by ViZiD — APPROVED** [review]

**Comment by s0me1newithhand7s** [comment]
ofborg sent:
```Shell
       > trying https://github.com/xvzc/SpoofDPI/archive/0.10.12.tar.gz
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
       For full logs, run 'nix-store -l /nix/store/4mhwd9k846msmx5531kq5nd2la02r848-source.drv'.
error: 1 dependencies of derivation '/nix/store/wknkgwmz7pyf5ir8zbc2yk0gy68ixcp2-SpoofDPI-0.10.12-go-modules.drv' failed to build
error: 2 dependencies of derivation '/nix/store/8py9f8bg203zvr3y83ijd6idfcjwi1d2-SpoofDPI-0.10.12.drv' failed to build
```

i'll run `nixpkgs-review` after work.

**Comment by s0me1newithhand7s** [comment]
`nixpkgs-review` built `spoofdpi` for `x86_64-linux`:
image

**Review by TheMaxMur — APPROVED** [review]

**Review by Pandapip1 — APPROVED** [review]
Looks pretty straightforward.

**Review by honnip — APPROVED** [review]

**Review by q60 — APPROVED** [review]

**Comment by s0me1newithhand7s** [comment]
i'll do a "update" draft PR bc newer version released.

**Comment by JohnRTitor** [comment]
> i'll do a "update" draft PR bc newer version released.

It's better to - Update the package here, squash the commits, then force push.

File: pkgs/by-name/sp/spoofdpi/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    hash = "sha256-GdGOnaIDy7XWWo0MOu+HfQcLrW/PDlRxf0y1jjJrZNQ=";
+  };
+
+  vendorHash = "sha256-47Gt5SI6VXq4+1T0LxFvQoYNk+JqTt3DonDXLfmFBzw=";
+
+  meta = {
+    homepage = "https://github.com/xvzc/SpoofDPI";
+    description = "A simple and fast anti-censorship tool written in Go";
+    license = lib.licenses.asl20;
+    platforms = lib.platforms.all;
```
Comment (JohnRTitor) [inline pkgs/by-name/sp/spoofdpi/package.nix:26]: Unset this, this is set by buildGoModule AFAIK

Comment (s0me1newithhand7s) [reply pkgs/by-name/sp/spoofdpi/package.nix:26]: i've took some pakcages written on go like `gat` and saw standart meta. :D

Comment (s0me1newithhand7s) [reply pkgs/by-name/sp/spoofdpi/package.nix:26]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/go/module.nix

eh

Comment (JohnRTitor) [reply pkgs/by-name/sp/spoofdpi/package.nix:26]: It's actually this

https://github.com/NixOS/nixpkgs/blob/91b1ab3465a334c2de084e6945238e88c2d1a302/pkgs/build-support/go/module.nix#L314

Comment (s0me1newithhand7s) [reply pkgs/by-name/sp/spoofdpi/package.nix:26]: > It's actually this
> 
> https://github.com/NixOS/nixpkgs/blob/91b1ab3465a334c2de084e6945238e88c2d1a302/pkgs/build-support/go/module.nix#L314

oh, ok!

**Review by JohnRTitor — COMMENTED** [review]
Cleanup the commit message, only the co-authored by line is enough.

**Comment by s0me1newithhand7s** [comment]
> Cleanup the commit message, only the co-authored by line is enough.

'k!

**Review by Pandapip1 — APPROVED** [review]
Diff still LGTM; untested

**Comment by s0me1newithhand7s** [comment]
> Diff still LGTM; untested

running test rn.

**Comment by Pandapip1** [comment]
Result of `nixpkgs-review pr 337795` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spoofdpi</li>
  </ul>
</details>

**Review by dr460nf1r3 — APPROVED** [review]

**Comment by s0me1newithhand7s** [comment]
TYSM, y'all!

File: pkgs/by-name/sp/spoofdpi/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  buildGoModule,
+  ...
+}:
+
+buildGoModule rec {
+  pname = "SpoofDPI";
```
Comment (SuperSandro2000) [inline pkgs/by-name/sp/spoofdpi/package.nix:10]: pnames should be lower cased

File: pkgs/by-name/sp/spoofdpi/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    repo = "SpoofDPI";
+    rev = "v${version}";
+    hash = "sha256-GdGOnaIDy7XWWo0MOu+HfQcLrW/PDlRxf0y1jjJrZNQ=";
+  };
+
+  vendorHash = "sha256-47Gt5SI6VXq4+1T0LxFvQoYNk+JqTt3DonDXLfmFBzw=";
+
+  meta = {
+    homepage = "https://github.com/xvzc/SpoofDPI";
+    description = "A simple and fast anti-censorship tool written in Go";
```
Comment (SuperSandro2000) [inline pkgs/by-name/sp/spoofdpi/package.nix:24]: descriptions should not start with an article

Comment (s0me1newithhand7s) [reply pkgs/by-name/sp/spoofdpi/package.nix:24]: i've copypasted description. :D

File: pkgs/by-name/sp/spoofdpi/package.nix (hunk 2 above)
Comment (s0me1newithhand7s) [reply pkgs/by-name/sp/spoofdpi/package.nix:10]: i'll change it with next update!

File: pkgs/by-name/sp/spoofdpi/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  buildGoModule,
+  ...
```
Comment (SuperSandro2000) [inline pkgs/by-name/sp/spoofdpi/package.nix:6]: This line is not necessary

Comment (s0me1newithhand7s) [reply pkgs/by-name/sp/spoofdpi/package.nix:6]: wdym?

**Pull request merged**